Conda environment file.

zwn opened the pull request:
I've started with conda environment file:
```yml
name: osgar
channels:
- conda-forge
- defaults
dependencies:
- certifi=2017.4.17=py27_0
- jpeg=9b=vc9_0
- libpng=1.6.28=vc9_0
- libtiff=4.0.6=vc9_7
- opencv=2.4.13=np113py27_1
- pip=9.0.1=py27_0
- python=2.7.13=1
- setuptools=33.1.1=py27_0
- vc=9=0
- wheel=0.29.0=py27_0
- wincertstore=0.2=py27_0
- zlib=1.2.11=vc9_0
- mkl=2017.0.1=0
- numpy=1.13.0=py27_0
- vs2008_runtime=9.00.30729.5054=0
```
but I am to entirely happy with the result. From the package names it is clear there are platform specific packages so this would be the win32 version of the environment.

But I guess this is as good as it gets (?)

Recreate on Windows 7 with
```
$ conda env create -f conda-win32.yml -n osgar
```

@m3d Let me know if it works for you.

Further work:
- [x] add linux version of the file (to be used on the robot)
- [x] add pyserial and pygame

zwn commented:
Oh, maybe conda is not such a good idea. For more info please see

https://github.com/conda/conda/issues/5036#issuecomment-309794842
https://github.com/conda/conda/issues/5435#issuecomment-309797922

But I don't have any alternatives.

zwn commented:
@m3d I've edited the command to create the environment to:
```
$ conda env create -f conda-win32.yml -n osgar
```
in the previous post.

m3d commented:
I would name branches with feature/ or fix/ prefix

m3d commented:
```
d:\md\git\osgar>conda env create -f conda-win32.yml -n osgar
Fetching package metadata .............
Solving package specifications: .
mkl-2017.0.1-0 100% |###############################| Time: 0:00:21   6.28 MB/s
vs2008_runtime 100% |###############################| Time: 0:00:00   9.82 MB/s
jpeg-9b-vc9_0. 100% |###############################| Time: 0:00:01 207.77 kB/s
python-2.7.13- 100% |###############################| Time: 0:00:11   1.74 MB/s
vc-9-0.tar.bz2 100% |###############################| Time: 0:00:00 411.98 kB/s
zlib-1.2.11-vc 100% |###############################| Time: 0:00:00 805.92 kB/s
certifi-2017.4 100% |###############################| Time: 0:00:00 778.72 kB/s
libpng-1.6.28- 100% |###############################| Time: 0:00:00 796.52 kB/s
libtiff-4.0.6- 100% |###############################| Time: 0:00:00 730.67 kB/s
numpy-1.13.0-p 100% |###############################| Time: 0:00:00   9.17 MB/s
wincertstore-0 100% |###############################| Time: 0:00:00   2.56 MB/s
opencv-2.4.13- 100% |###############################| Time: 0:00:12   1.69 MB/s
setuptools-33. 100% |###############################| Time: 0:00:00   2.65 MB/s
wheel-0.29.0-p 100% |###############################| Time: 0:00:00   8.90 MB/s
pip-9.0.1-py27 100% |###############################| Time: 0:00:01   1.44 MB/s
#
# To activate this environment, use:
# > activate osgar
#
# To deactivate this environment, use:
# > deactivate osgar
#
# * for power-users using bash, you must source
#
```
so installation was without problem, but I confirm that even for replay of log files you need pyserial:
```
(osgar) d:\md\git\osgar>python navpat.py d:\md\osgar\logs\czu170613\czu1\meta_17
0613_180934.log
Traceback (most recent call last):
  File "navpat.py", line 15, in <module>
    from can import CAN, DummyMemoryLog, ReplayLogInputsOnly, ReplayLog
  File "d:\md\git\osgar\can.py", line 8, in <module>
    import serial
ImportError: No module named serial
```

zwn commented:
> I would name branches with feature/ or fix/ prefix

Well, I am not sure when something is a fix and when it is already a feature. What would we need that distinction for?

zwn commented:
Ticketed as #11

zwn commented:
I have rebased the branch on top of feature/improve-navpat since that is compatible with the latest logfiles. Otherwise I have no option how to test it.

I've added pygame and pyserial. Local development on Windows 7 should be covered now. One task left - do the same for linux.

@m3d Please test again.

zwn commented:
@m3d If everything runs well inside these environments, it would be great if you could start using it locally as well as on the robot. I'd like to base automatic tests on them.

m3d commented:
Please do not rebase while pull request is "in progress". Use fixup commits instead with final rebase at the end.

m3d commented:
```
martind@lenovo2 MINGW32 /d/md/git/osgar ((158213c...))
$ python ./navpat.py

  Navigate given pattern in selected area
  usage:
       ./navpat.py <notes> | [<metalog> [<F>]] [--view]

(osgar2)
```
+ replay of 
```
$ python ./navpat.py /d/md/osgar/logs/czu170530/czu3/meta_170530_170246.log
```
passed fine :+1:

zwn commented:
@m3d I'll wait with the merge since you've said you are going to break up the "improvement" branch into smaller ones. Please start with replacing the submodule appyros since it is the most annoying to switch to/from.
